FirebaseInAppMessaging/Swift/Tests/Unit/CustomInAppMessageDisplayViewModifierTests.swift

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+
+@testable import FirebaseInAppMessaging
+@testable import FirebaseInAppMessagingSwift
+
+class DelegateBridgeTests: XCTestCase {
+  var delegateBridge = DelegateBridge()
+
+  func testClearsInAppMessageOnDismiss() {
+    let inAppMessage: InAppMessagingDisplayMessage = InAppMessagingDisplayMessage(
+      messageID: "messageID",
+      campaignName: "testCampaign",
+      renderAsTestMessage: false,
+      messageType: .card,
+      triggerType: .onAppForeground
+    )
+    delegateBridge.displayMessage(inAppMessage, displayDelegate: TestDelegate())
```

Comment:
Can we verify anything on the  `TestDelegate()`  here?

Reply:
@ryanwilson The only thing we're missing is testing that the impression fires in `onAppear()` see line 34 of CustomInAppMessageDisplayViewModifier.swift for example. I don't see a way of doing this without opening up the initializers for the other message subclasses (since it involves mocking the entire flow). Let's table this and handle it when we handle UI testing for the integration test app.